core/shared/src/main/scala-2.13/fs2/ChunkPlatform.scala

```diff
@@ -0,0 +1,41 @@
+package fs2
+
+import scala.collection.immutable.ArraySeq
+import scala.collection.{immutable, mutable}
+import scala.reflect.ClassTag
+
+trait ChunkPlatform[+O] { self: Chunk[O] =>
+
+  def toArraySeq[O2 >: O: ClassTag]: ArraySeq[O2] = {
+    val array: Array[O2] = new Array[O2](size)
+    copyToArray(array)
+    ArraySeq.unsafeWrapArray[O2](array)
```

Comment:
This seems to be the most efficient way to construct an `ArraySeq`, given it will often take advantage of `System.arraycopy`. I might be missing something, though, so I'm open to feedback if there are better ways to do this.